include/meshio/vectors.hpp

```diff
@@ -1,178 +1,205 @@
 /*
- * Copyright (c) 2015, Lakshman Anumolu, Pradeep Garigipati
+ * Copyright (c) 2017, Lakshman Anumolu, Pradeep Garigipati
  * All rights reserved.
  *
  * This file is part of MeshIO whose distribution is governed by
  * the BSD 2-Clause License contained in the accompanying LICENSE.txt
  * file.
  */
 
-#ifndef __VECTORS_HPP__
-#define __VECTORS_HPP__
+#ifndef MESHIO_VECTORS_HPP_
+#define MESHIO_VECTORS_HPP_
 
+#include <cstdlib>
 #include <vector>
 
 namespace meshio
 {
 
 template<class T>
-class Vec2 {
+class Vec4 {
   public:
-    T x, y;
-
-    Vec2(T pX, T pY)
-        : x(pX), y(pY) {
-    }
-
-    Vec2()
-        : x(0), y(0) {
-    }
-
-    Vec2& operator+=(const Vec2& other) {
-        x += other.x;
-        y += other.y;
-        return *this;
-    }
-
-    Vec2& operator-=(const Vec2& other) {
-        x -= other.x;
-        y -= other.y;
-        return *this;
-    }
-
-    Vec2& operator*=(const T pDiv) {
-        x *= pDiv;
-        y *= pDiv;
-        return *this;
-    }
-
-    Vec2& operator/=(const T pDiv) {
-        T oneByPDiv = (T)1/pDiv;
-        return operator*=(oneByPDiv);
-    }
-
-    bool operator==(const Vec2& other) {
-        return ((x == other.x) && (y == other.y));
-    }
+    typedef typename std::vector<T>::iterator viterator;
+    typedef typename std::vector<T>::const_iterator const_viterator;
+
+    Vec4(T pX, T pY, T pZ, T pW) : elems({pX, pY, pZ, pW}) {}
+    Vec4() : Vec4(0, 0, 0, 1) { } /* TODO: In graphics applications we
+                                     generally use fourth point as 1.
+                                     If this is not OK we can change it */
```

Comment:
I think that is fine especially since we do provide constructor where the user can set the vector components explicitly to which ever values he wants to.